rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -207,6 +207,80 @@ public void getRecoveringLRAs(Map<URI, Transaction> lras) {
         }
     }
 
+    // remove a log record
+    public boolean removeCommitted(URI lraId) {
+        InputObjectState aa_uids = new InputObjectState();
+
+        try {
+            if (getUids(aa_uids)) {
+                boolean moreUids = true;
+
+                while (moreUids) {
+                    try {
+                        Uid uid = UidHelper.unpackFrom(aa_uids);
+
+                        if (uid.equals(Uid.nullUid())) {
+                            moreUids = false;
+                        } else {
+                            Transaction lra = new Transaction(lraService, new Uid(uid));
+
+                            lra.activate(); // or just parse lraId extracting the uid
+
+                            if (uid.fileStringForm().equals(lra.getUid())) {
+                                return _recoveryStore.remove_committed(uid, lra.type());
+                            }
+                        }
+                    } catch (Exception ex) {
+                        moreUids = false;
+                    }
+                }
+            }
+        } catch (ObjectStoreException e) {
+            if (LRALogger.logger.isInfoEnabled()) {
+                LRALogger.logger.infof("LRARecoverModule: Object store exception: %s", e.getMessage());
+            }
+        }
+
+        return false;
+    }
+
+    public void getFailedLRAs(Map<URI, Transaction> lras) {
+        InputObjectState aa_uids = new InputObjectState();
+
+        try {
+            if (getUids(aa_uids)) {
+                boolean moreUids = true;
+
+                while (moreUids) {
+                    try {
+                        Uid uid = UidHelper.unpackFrom(aa_uids);
+
+                        if (uid.equals(Uid.nullUid())) {
+                            moreUids = false;
+                        } else {
+                            Transaction lra = new Transaction(lraService, new Uid(uid));
+
+                            lra.activate();
+                            lras.put(lra.getId(), lra);
+                        }
+                    } catch (Exception ex) {
+                        moreUids = false;
```

Comment:
Could we log the exception reason and the stacktrace under TRACE log level, please?

Reply:
+1

Reply:
Fixed with the last commit